libs/ui-lib/lib/ocm/services/apis/NewFeatureSupportLevelsAPI.ts

```diff
@@ -1,26 +1,24 @@
-import {
-  ArchitectureSupportLevelMap,
-  NewFeatureSupportLevelMap,
-} from '../../../common/components/newFeatureSupportLevels';
+import { CpuArchitecture } from '../../../common';
 import { clientWithoutConverter } from '../../api/axiosClient';
-
-export interface ArchitecturesSupportsLevel {
-  architectures: ArchitectureSupportLevelMap;
-}
-
-export interface FeaturesSupportsLevel {
-  features: NewFeatureSupportLevelMap;
-}
+import {
+  ArchitecturesSupportsLevel,
+  FeaturesSupportsLevel,
+} from '../../components/newFeatureSupportLevels/types';
+import { mapOcmArchToCpuArchitecture } from '../CpuArchitectureService';
 
 const NewFeatureSupportLevelsAPI = {
   makeBaseURI() {
     return `/v2/support-levels`;
   },
 
-  featuresSupportLevel(openshiftVersion: string, cpuArchitecture?: string) {
+  featuresSupportLevel(openshiftVersion: string, cpuArchitecture?: string, isOcm?: boolean) {
+    const cpuArch =
```

Comment:
This kind of logic should be performed elsewhere not here. The *API objects are just wrapping the API calls.
If this piece of logic needs to be reused consider creating a service